core/src/data_sources/data_source.rs

```diff
@@ -1717,6 +1746,64 @@ impl DataSource {
         Ok(())
     }
 
+    pub async fn scrub_document_deleted_versions(
```

Comment:
Aren't we worried that this hand up for too long? Can't we delete the overall folder in one call since they use the same prefix?

Pretty sure that we have some documents coming from managed data sources that have been updated a few hundreds of times...

Reply:
We don't want to delete the folder since we might still have working versions right?

If the production check that scrubs works well we should have little outstanding deleted versions right? It appears that we have a few but I think that's fine we might have a few long calls early on and then since we added the scrubbing as part of the delete we will be able to remove the production check entirely right?

Reply:
Good to me.